tests/unittest_inference.py

```diff
@@ -3768,6 +3768,33 @@ class MetaBook(type):
         ]
         self.assertEqual(titles, ["Catch 22", "Ubik", "Grimus"])
 
+    @staticmethod
+    def test_builtin_new() -> None:
+        ast_node = extract_node("int.__new__(int, 42)")
+        inferred = next(ast_node.infer())
+        assert isinstance(inferred, nodes.Const)
+        assert inferred.value == 42
+
+        ast_node2 = extract_node("int.__new__(int)")
+        inferred2 = next(ast_node2.infer())
+        assert isinstance(inferred2, Instance)
+        assert not isinstance(inferred2, nodes.Const)
```

Comment:
Isn't the second assert redundant? If it is an `Instance`, it's not a `Const`.

Reply:
Actually `Const` subclasses `Instance`. Trying to do it all in one assertion with `issubclass` seemed less straightforward to me.